### PR TITLE
Replace NotNullAttribute with thrown exceptions

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNet.Session/DistributedSession.cs [hunk 1]
```diff
@@ -56,8 +80,13 @@ public bool TryGetValue(string key, out byte[] value)
             return _store.TryGetValue(new EncodedKey(key), out value);
         }
 
-        public void Set(string key, [NotNull] byte[] value)
+        public void Set(string key, byte[] value)
         {
+            if (value == null)
```
Comment (hishamco) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: Shall we check if the ```value == 0``` too

**Comment by hishamco** [comment]
:+1:

File: src/Microsoft.AspNet.Session/DistributedSession.cs [hunk 2]
```diff
@@ -29,9 +28,34 @@ public class DistributedSession : ISession
...
+            Func<bool> tryEstablishSession,
+            ILoggerFactory loggerFactory,
+            bool isNewSessionKey)
         {
+            if (cache == null)
+            {
+                throw new ArgumentNullException(nameof(cache));
+            }
+
+            if (sessionId == null)
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: IsNullOrEmpty

File: src/Microsoft.AspNet.Session/DistributedSession.cs (hunk 1 above)
Comment (Tratcher) [inline src/Microsoft.AspNet.Session/DistributedSession.cs]: No, this is ok.

File: src/Microsoft.AspNet.Session/DistributedSessionStore.cs [hunk 3]
```diff
@@ -33,8 +42,18 @@ public void Connect()
             _cache.Connect();
         }
 
-        public ISession Create([NotNull] string sessionId, TimeSpan idleTimeout, [NotNull] Func<bool> tryEstablishSession, bool isNewSessionKey)
+        public ISession Create(string sessionId, TimeSpan idleTimeout, Func<bool> tryEstablishSession, bool isNewSessionKey)
         {
+            if (sessionId == null)
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Session/DistributedSessionStore.cs]: IsNullOrEmpty

**Comment by Tratcher** [comment]
:shipit:

**Pull request closed without merging**